### PR TITLE
[[ Bug 20000 ]] Scriptify sQuiRt QR library

### PR DESCRIPTION
**Pull request by montegoulding**

### CODE REVIEW

File: extensions/script-libraries/qr/qr.livecodescript [hunk 1]
```diff
@@ -0,0 +1,2443 @@
...
+      if there is an img pImg then
+         put the long id of img pImg into pImg
+         put false into tPNG
+      else
+         put not exists(pImg) into tPNG
+      end if
+      
+      # check ECC
+      if not(pECC is among the items of "L,M,Q,H") then
+         return "Error: ECC must be L, M, Q or H"
```
Comment (mwieder) [inline extensions/script-libraries/qr/qr.livecodescript]: Is this legal? (returning while in a try construct?) Shouldn't it rather throw the error which will then be caught by the catch clause and be handled gracefully?

Comment (montegoulding) [reply extensions/script-libraries/qr/qr.livecodescript]: It's fine but you could get some interesting result if there's a finally clause as that will be executed still.

Comment (mwieder) [reply extensions/script-libraries/qr/qr.livecodescript]: It's already going to miss the 'unlock screen' at the end. I'd feel better about changing this to
throw "ECC must be L, M, Q or H"

Comment (montegoulding) [reply extensions/script-libraries/qr/qr.livecodescript]: Good catch.. yeah it's probably a good idea to change to throw. It's  also quite likely the lock screen could just wrap the image setting code.

**Comment by livecodeali** [comment]
@livecode-vulcan review ok b01d535

**Comment by livecode-vulcan** [comment]
💙 review by @livecodeali ok b01d535862c88c4dd46c3976215b0f68a0aeef3b

**Comment by livecode-vulcan** [comment]
😎 test success b01d535862c88c4dd46c3976215b0f68a0aeef3b

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/488)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/503)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/475)
* try-community-universal-ios-iphoneos10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos10.3/builds/418)
* try-community-universal-ios-iphonesimulator10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator10.3/builds/421)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/483)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/489)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/486)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/501)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/486)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/496)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/513)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/481)

**Pull request merged**